src/chapters/unit-testing/exercises.rst

```diff
@@ -0,0 +1,326 @@
+
+   a. In the ``checkFive.spec.js`` file, add an empty ``describe`` function:
+
+      .. sourcecode:: js
+
+         const test = require('../checkFive.js');
+         const assert = require('assert');
+
+         describe("checkFive", function(){
+
+            //testing code here...
+
+         });
+
+   b. Now add an ``it`` clause.
+
+      .. sourcecode:: js
+
+         const test = require('../checkFive.js');
+         const assert = require('assert');
+
+         describe("checkFive", function(){
+
+            it("Descriptive feedback...", function(){
+               //code here...
+            });
+
+         });
+
+#. Let's write a test to see if ``checkFive`` produces the correct output when
+   passed a number *less than 5*.
+
+   a. First, replace ``Descriptive feedback...`` with a DETAILED message. This
+      is the text that the user will see if the test *fails*. Do NOT skimp on
+      this. Best practice suggests something like ``'should [DO THIS] if [THIS
+      HAPPENS]'``, where the text inside the ``[]`` describes what the function
+      should do when passed a specific input.
+   b. Define the variable ``output``, and initialize it by passing a value of
+      ``2`` to ``checkFive``.
+
+      .. sourcecode:: js
+
+         const test = require('../checkFive.js');
+         const assert = require('assert');
+
+         describe("checkFive", function(){
+
+            it("Descriptive feedback...", function(){
+               let output = test.checkFive(2);
+            });
+
+         });
+
+   c. Now use the ``assert`` function to check the result:
+
+      .. sourcecode:: js
+
+         const test = require('../checkFive.js');
+         const assert = require('assert');
+
+         describe("checkFive", function(){
+
+            it("Descriptive feedback...", function(){
+               let output = test.checkFive(2);
+               assert.equal(output, "2 is less than 5.");
+            });
+
+         });
+
+   d. Run the test script and examine the results. The test should have
+      passed and prodced output similar to:
+
+      ::
+
+         Started
+         .
+
+         1 spec, 0 failures
+         Finished in 0.006 seconds
+
+   e. Now change line 3 in ``checkFive.js`` to ``if (num > 5)`` and rerun
+      the test. The output should look similar to :
+
+      ::
+
+         Started
+         F
+
+         Failures:
+         1) checkFive should return 'num is less than 5.' if given a number less than 5
+         Message:
+            AssertionError [ERR_ASSERTION]: '2 is greater than 5.' == '2 is less than 5.'
+
+   f. Change line 3 back.
+
+   .. note::
+
+      We do NOT need to check every possible value that is less than 5. Testing a single
+      example is sufficient to check that part of the function.
+
+#. Add two more ``it`` clauses inside ``describe`` to test what happens when
+   ``checkFive`` is passed a value greater than 5 or equal to 5.
+
+
+Try One on Your Own
```

Comment:
I really like this problem, however what the student needs to do needs to be explained more. We should flat out tell them to use unit tests to find the bug in the program. Tell them to write out test cases. Then write tests for each case. Then fix the bug base the test results.